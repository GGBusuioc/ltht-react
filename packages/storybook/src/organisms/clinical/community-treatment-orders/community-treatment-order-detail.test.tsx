import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import CommunityTreatmentOrderDetail from '@ltht-react/community-treatment-order-detail'
import orders from './community-treatment-order.fixtures'

describe('CommunityTreatmentOrderSummary', () => {
  const communityTreatmentOrder = orders[0]
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<CommunityTreatmentOrderDetail communityTreatmentOrder={communityTreatmentOrder} />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<CommunityTreatmentOrderDetail communityTreatmentOrder={communityTreatmentOrder} />)).toMatchSnapshot(
      'wrapper mount'
    )
  })
})
