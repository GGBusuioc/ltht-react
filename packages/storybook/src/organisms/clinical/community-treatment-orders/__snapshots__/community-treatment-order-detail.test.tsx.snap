// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityTreatmentOrderSummary matches snapshot: wrapper mount 1`] = `
.emotion-2 {
  list-style: none;
  margin-top: 1rem !important;
}

.emotion-0 {
  color: #98a4ad !important;
  margin-bottom: 0.25rem !important;
}

.emotion-1 {
  color: -v-a-l-u-e:#768692;
  -d-a-r-k-e-r25: #59656e;
  -l-i-g-h-t-e-r25: #98a4ad;
}

<CommunityTreatmentOrderDetail
  communityTreatmentOrder={
    Object {
      "consentToTreat": "Consent Not Given",
      "id": "580ca927-34e0-e911-a2c7-005056926fe4|communityTreatmentOrder",
      "legalStatus": Object {
        "coding": Array [
          Object {
            "code": "3309007",
            "display": "Approved",
          },
        ],
        "text": "Approved",
      },
      "metadata": Object {
        "dataSources": Array [
          Object {
            "display": "Mock",
          },
        ],
        "isRedacted": false,
        "requestedWhen": "",
      },
      "period": Object {
        "end": Object {
          "kind": "DATE",
          "value": "2014-02-02T00:00:00+00:00",
        },
        "start": Object {
          "kind": "DATE",
          "value": "2013-01-01T00:00:00+00:00",
        },
      },
      "restrictions": "No restrictions",
    }
  }
>
  <StringDetail
    description="Consent Not Given"
    term="Consent"
  >
    <DescriptionList>
      <dl
        className="emotion-2"
      >
        <DescriptionListTerm>
          <dt
            className="emotion-0"
          >
            Consent
          </dt>
        </DescriptionListTerm>
        <DescriptionListDescription>
          <dd
            className="emotion-1"
          >
            Consent Not Given
          </dd>
        </DescriptionListDescription>
      </dl>
    </DescriptionList>
  </StringDetail>
  <CodeableConceptDetail
    concept={
      Object {
        "coding": Array [
          Object {
            "code": "3309007",
            "display": "Approved",
          },
        ],
        "text": "Approved",
      }
    }
    term="Legal Status"
  >
    <DescriptionList>
      <dl
        className="emotion-2"
      >
        <DescriptionListTerm>
          <dt
            className="emotion-0"
          >
            Legal Status
          </dt>
        </DescriptionListTerm>
        <DescriptionListDescription>
          <dd
            className="emotion-1"
          >
            Approved
          </dd>
        </DescriptionListDescription>
      </dl>
    </DescriptionList>
  </CodeableConceptDetail>
  <PeriodDetail
    period={
      Object {
        "end": Object {
          "kind": "DATE",
          "value": "2014-02-02T00:00:00+00:00",
        },
        "start": Object {
          "kind": "DATE",
          "value": "2013-01-01T00:00:00+00:00",
        },
      }
    }
  >
    <DescriptionList>
      <dl
        className="emotion-2"
      >
        <DescriptionListTerm>
          <dt
            className="emotion-0"
          >
            Period
          </dt>
        </DescriptionListTerm>
        <DescriptionListDescription>
          <dd
            className="emotion-1"
          >
            Jan-01,-2013 to Feb-02,-2014
          </dd>
        </DescriptionListDescription>
      </dl>
    </DescriptionList>
  </PeriodDetail>
  <StringDetail
    description="No restrictions"
    term="Restrictions"
  >
    <DescriptionList>
      <dl
        className="emotion-2"
      >
        <DescriptionListTerm>
          <dt
            className="emotion-0"
          >
            Restrictions
          </dt>
        </DescriptionListTerm>
        <DescriptionListDescription>
          <dd
            className="emotion-1"
          >
            No Restrictions
          </dd>
        </DescriptionListDescription>
      </dl>
    </DescriptionList>
  </StringDetail>
</CommunityTreatmentOrderDetail>
`;
